src/lib/block_producer/block_producer.ml

```diff
@@ -736,6 +787,21 @@ let run ~logger ~prover ~verifier ~trust_system ~get_completed_work
                   Transition_frontier.best_tip transition_frontier
                   |> Breadcrumb.consensus_state
                 in
+                let generate_genesis_proof_if_needed () =
+                  match Broadcast_pipe.Reader.peek frontier_reader with
+                  | Some transition_frontier ->
+                      let consensus_state =
+                        Transition_frontier.best_tip transition_frontier
+                        |> Breadcrumb.consensus_state
+                      in
+                      if
+                        Consensus.Data.Consensus_state.is_genesis_state
+                          consensus_state
```

Comment:
> A node when restarted with clean config directory will have genesis state as its best tip. Might be better to not produce a block at all in this case.

I agree, but I think this falls outside of the scope of this PR. We should probably avoid checking vrfs/running the `start` function at all until we have an epoch ledger from a block that's within the current epoch, which would fix a lot of the reported bp weirdness during startup.

Currently, this code will only trigger if they've won a slot that's imminent (this or the next) but haven't been running for long enough to see their best tip progress. If it's this slot, they would produce a useless block on top of the genesis block anyway (waiting to produce it may even allow them time to get a plausible best tip); if it's the next, they'll waste some compute generating a genesis proof that currently every node is wasting at startup regardless.